drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -0,0 +1,182 @@
+{
+	const int pkt_len = net_pkt_get_len(pkt);
+
+	/* Read the packet payload */
+	if (net_pkt_read(pkt, DEV_DATA(dev)->frame_buf, pkt_len) < 0) {
+		return -EIO;
+	}
+
+	/* Enqueue packet for transmission */
+	esp_wifi_internal_tx(ESP_IF_WIFI_STA, (void *)DEV_DATA(dev)->frame_buf, pkt_len);
+
+	LOG_DBG("pkt sent %p len %d", pkt, pkt_len);
+
+	return 0;
+}
+
+esp_err_t eth_esp32_rx(void *buffer, uint16_t len, void *eb)
+{
+	struct net_pkt *pkt;
+	struct net_if *iface;
+
+	iface = net_if_get_first_by_type(&NET_L2_GET_NAME(ETHERNET));
+	pkt = net_pkt_rx_alloc_with_buffer(iface, len,
+			AF_UNSPEC, 0, K_NO_WAIT);
+	if (!pkt) {
+		LOG_ERR("Failed to get net buffer");
+		return ESP_FAIL;
+	}
+
+	if (net_pkt_write(pkt, buffer, len) < 0) {
+		LOG_ERR("Failed to write pkt");
+		goto pkt_unref;
+	}
+
+	if (net_recv_data(iface, pkt) < 0) {
+		LOG_ERR("Failed to push received data");
+		goto pkt_unref;
+	}
+	esp_wifi_internal_free_rx_buffer(eb);
+	return ESP_OK;
+
+pkt_unref:
+	net_pkt_unref(pkt);
+	return ESP_FAIL;
+}
+
+void esp_wifi_set_net_state(bool state)
+{
+	struct net_if *iface;
+
+	iface = net_if_get_first_by_type(&NET_L2_GET_NAME(ETHERNET));
+	if (iface == NULL) {
+		LOG_ERR("network interface unavailable");
+		return;
+	}
+
+	if (state) {
+		esp_wifi_internal_reg_rxcb(ESP_IF_WIFI_STA, (wifi_rxcb_t)eth_esp32_rx);
+		net_if_up(iface);
+	} else {
+		esp_wifi_internal_reg_rxcb(ESP_IF_WIFI_STA, NULL);
+		net_if_down(iface);
+	}
+}
+
+static void eth_esp32_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct esp32_wifi_runtime *dev_data = DEV_DATA(dev);
+
+	dev_data->iface = iface;
+	/* Start interface when we are actually connected with WiFi network */
+	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+	esp_read_mac(dev_data->mac_addr, ESP_MAC_WIFI_STA);
+
+	/* Assign link local address. */
+	net_if_set_link_addr(iface,
+			dev_data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	ethernet_init(iface);
+}
+
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+static struct net_stats_eth *eth_esp32_stats(const struct device *dev)
+{
+	return &(DEV_DATA(dev)->stats);
+}
+#endif
+
+static int eth_esp32_dev_init(const struct device *dev)
+{
+	esp_timer_init();
+	esp_event_init();
+
+	wifi_init_config_t config = WIFI_INIT_CONFIG_DEFAULT();
+	esp_err_t ret = esp_wifi_init(&config);
+
+	ret |= esp_supplicant_init();
+	ret |= esp_wifi_start();
+
+	if (IS_ENABLED(CONFIG_WIFI_STA_AUTO)) {
+		wifi_config_t wifi_config = {
+			.sta = {
+				.ssid = CONFIG_WIFI_SSID,
+				.password = CONFIG_WIFI_PASSWORD,
+			},
+		};
+
+		ret = esp_wifi_set_mode(WIFI_MODE_STA);
+		ret |= esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
+		ret |= esp_wifi_connect();
+	}
+
+	if (ret != ESP_OK) {
+		LOG_ERR("Connect failed");
+	}
+
+	return ret;
+}
+
+DEVICE_DECLARE(eth_esp32);
+
+struct esp32_wifi_runtime eth_data = {
+	.mac_addr = {0},
+	.tx_err = false,
+	.tx_word = 0,
+	.tx_pos = 0,
+};
```

Comment:
static?

All members of this structure will be initialized to 0, so I see no point in explicit initialization.